src/Workspaces/Core/Portable/Diagnostics/DiagnosticAnalysisResultBuilder.cs

```diff
@@ -113,7 +113,7 @@ public void AddExternalSemanticDiagnostics(DocumentId documentId, IEnumerable<Di
 
         private void AppendDiagnostics(ref Dictionary<DocumentId, List<DiagnosticData>> map, Document documentOpt, Diagnostic diagnostic)
         {
-            if (documentOpt?.SupportsDiagnostics() == false)
+            if (documentOpt?.SupportsDiagnostics() ?? false)
```

Comment:
I am a bit confused: are we returning if `documentOpt?.SupportsDiagnostics()` returns `true`? Honestly, I just prefer breaking this into `if (documentOpt == null || !documentOpt.SupportsDiagnostics())` instead of having both a conditional access and a coalesce expression in an expression.

Reply:
Sigh...

Reply:
Fixed